File: C2103L/IASF/bookapp/src/main/java/com/aptech/bookapp/services/IUserService.java
package com.aptech.bookapp.services;

import com.aptech.bookapp.models.User;
import com.aptech.bookapp.viewmodels.UserRegisterModel;
import org.springframework.stereotype.Service;

@Service
public interface IUserService {
    User login(String username, String password);
    User register(UserRegisterModel model);
}

File: C2108G2/DMA/myapp/Services/User/IUserService.cs
using System;
using myapp.Models;
using myapp.ViewModels;

namespace myapp.Services
{
    public interface IUserService
    {
        Task<string> Authenticate(LoginViewModel loginViewModel);
        Task<User?> Register(RegisterViewModel registerViewModel);
        Task<User?> GetUserById(int id);
    }
}

